AdminCore.Services/Configuration/ConfigurationProvider.cs

```diff
@@ -21,5 +22,10 @@ public MigrationTypes RetrieveMigrationType()
 
       return MigrationTypes.Migrate;
     }
+
+    public IMailServerConfiguration RetrieveMailServiceConfig()
+    {
+      return new SmtpServerConfiguration();
```

Comment:
Could you skip adding a RetreiveMailServerConfig() method to IConfiguration and just bind in the DIC:
```
services.AddTransient<IMailServerConfiguration, SmtpServerConfiguration>();
```
This also means you aren't instantiating it directly in the ConfigurationProvider.

Then inject the IMailServerConfiguration into the constructor of the classes that depend on it?

Reply:
why transient for a configuration object?

Reply:
To replicate the functionality of your RetrieveMailServiceConfig() method, it returns a new instance of SmtpServerConfiguration every time it's called. A singleton might make more sense though.